agent/base_agent.py

```diff
@@ -29,18 +30,15 @@ def __init__(self,
     self.name: str = name or self.__class__.__name__
     self.dialog: str = ''  # Communication history between LLM and tool.
 
-    # TODO(dongge): Replace this with google-cloud-log in a module.
-    logging.basicConfig(level=logging.DEBUG,
-                        format=('%(asctime)s [Trial: %02d] %(levelname)s '
-                                '[%(module)s.%(funcName)s]: %(message)s'))
-
-    self.logger = logging.getLogger(__name__)
-    self.logger.setLevel(logging.DEBUG)
-
   def write_to_file(self, file_path: str, file_content: str):
     with open(file_path, 'w') as file:
       file.writelines(file_content)
 
+  def log(self, msg: str, level=logging.INFO):
```

Comment:
nit: can we keep the format string mechanism for logging? 

i.e. add `*args, **kwargs` here and let callers do `self.log('foo %s', 'arg')`.

Reply:
Yep, I pushed this as a temporary solution to show some quick results.
Ideally, I want to avoid manually defining this function and will think of a better solution.